app/src/main/java/protect/card_locker/PkpassParser.kt

```diff
@@ -0,0 +1,404 @@
+        }
+
+        val inputStream = mContext.contentResolver.openInputStream(uri) ?: throw IOException(mContext.getString(R.string.errorReadingFile))
+        val zipInputStream = ZipInputStream(inputStream)
+
+        try {
+            var localFileHeader: LocalFileHeader
+            while ((zipInputStream.nextEntry.also { localFileHeader = it }) != null) {
+                // Ignore directories
+                if (localFileHeader.isDirectory) continue
+
+                // We assume there are three options, as per spec:
+                // language.lproj/pass.strings
+                // file.extension
+                // More directories are ignored
+                val filenameParts = localFileHeader.fileName.split('/')
+                if (filenameParts.size > 2) {
+                    continue
+                } else if (filenameParts.size == 2) {
+                    // Doesn't seem like a language directory, ignore
+                    if (!filenameParts[0].endsWith(".lproj")) continue
+
+                    val locale = filenameParts[0].removeSuffix(".lproj")
+
+                    translations[locale] = parseLanguageStrings(ZipUtils.read(zipInputStream))
+                }
+
+                // Not a language, parse as normal files
+                when (localFileHeader.fileName) {
+                    "logo.png" -> loadImageIfBiggerSize(1, zipInputStream)
+                    "logo@2x.png" -> loadImageIfBiggerSize(2, zipInputStream)
+                    "logo@3x.png" -> loadImageIfBiggerSize(3, zipInputStream)
+                    "pass.json" -> passContent = ZipUtils.readJSON(zipInputStream) // Parse this last, so we're sure we have all language info
+                }
+            }
+
+            checkNotNull(passContent) { "File lacks pass.json" }
+        } catch (e: Exception) {
+            throw e
+         } finally {
+            try {
+                inputStream.close()
+            } catch (e: IOException) {
+                Log.e(TAG, "Error closing inputStream", e)
+            }
+
+            try {
+                zipInputStream.close()
+            } catch (e: IOException) {
+                Log.e(TAG, "Error closing zipInputStream", e)
+            }
+        }
+    }
+
+    fun listLocales(): List<String> {
+        return translations.keys.toList()
+    }
+
+    fun toLoyaltyCard(locale: String?): LoyaltyCard {
+        parsePassJSON(checkNotNull(passContent) { "Pkpass instance not yet initialized!" }, locale)
+
+        return LoyaltyCard(
+            -1,
+            store,
+            note,
+            validFrom,
+            expiry,
+            balance,
+            balanceType,
+            cardId,
+            barcodeId,
+            barcodeType,
+            headerColor,
+            starStatus,
+            lastUsed,
+            zoomLevel,
+            archiveStatus
+        )
+    }
+
+    fun getImage(): Bitmap? {
+        return image
+    }
```

Comment:
You can change the variable to specify a getter directly like `private val image: Bitmap get() = <load_something>`. This will behave as a function and you get the benefit of nice syntax.

Reply:
My main concern is that I don't want the variable to be set from the outside, but I do want it readable from the outside. I guess your suggestion here is to rename the internal variable away from "image"?